src/test/java/wolox/training/controllers/UserControllerTest.java

```diff
@@ -0,0 +1,166 @@
+    @MockBean
+    private BookRepository mockBookRepository;
+
+    private ObjectMapper objectMapper;
+
+    @PostConstruct
+    private void objectMapper() {
+        objectMapper = new ObjectMapper()
+                .registerModule(new ParameterNamesModule())
+                .registerModule(new Jdk8Module())
+                .registerModule(new JavaTimeModule());
+    }
+
+    @Before
+    public void setUp() {
+        bookTest = new Book();
+        bookTest.setAuthor("DC: Alan Moore");
+        bookTest.setImage("image.png");
+        bookTest.setTitle("Batman: The Killing Joke");
+        bookTest.setSubtitle("A subtitle");
+        bookTest.setPublisher("Publisher");
+        bookTest.setGenre("novela grafica");
+        bookTest.setYear("1988");
+        bookTest.setPages(100);
+        bookTest.setIsbn("ABCD1234");
+        userTest = new User();
+        userTest.setUsername("marti");
+        userTest.setName("Martina");
+        userTest.setBirthdate(LocalDate.parse("1993-12-09"));
+        userTest.addBook(bookTest);
+        users = new ArrayList<User>();
+    }
+
+    @Test
+    public void whenCreateAUser_thenUserIsPersisted() throws Exception {
+        mvc.perform(post("/api/users/").contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(userTest)))
+                .andExpect(status().isCreated());
+    }
+
+    @Test
+    public void whenGetAllUsers_thenReturnAListOfUsers() throws Exception {
+        users.add(userTest);
+        Mockito.when(mockUserRepository.findAll()).thenReturn(users);
+        mvc.perform(get("/api/users/").contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(content().json("[{\"id\":null,\"username\":\"marti\",\"name\":\"Martina\",\"birthdate\":\"1993-12-09\",\"books\":[{\"id\":null,\"genre\":\"novela grafica\",\"author\":\"DC: Alan Moore\",\"image\":\"image.png\",\"title\":\"Batman: The Killing Joke\",\"subtitle\":\"A subtitle\",\"publisher\":\"Publisher\",\"year\":\"1988\",\"pages\":100,\"isbn\":\"ABCD1234\"}]}]"));
+    }
+
+    @Test
+    public void whenFindAUserById_thenReturnTheUser() throws Exception {
+        Mockito.when(mockUserRepository.findById(1L)).thenReturn(ofNullable(userTest));
+        mvc.perform(get("/api/users/1").contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(content().json("{\"id\":null,\"username\":\"marti\",\"name\":\"Martina\",\"birthdate\":\"1993-12-09\",\"books\":[{\"id\":null,\"genre\":\"novela grafica\",\"author\":\"DC: Alan Moore\",\"image\":\"image.png\",\"title\":\"Batman: The Killing Joke\",\"subtitle\":\"A subtitle\",\"publisher\":\"Publisher\",\"year\":\"1988\",\"pages\":100,\"isbn\":\"ABCD1234\"}]}"));
+    }
+
+    @Test
+    public void whenFindAUserByUsername_thenReturnTheUser() throws Exception {
+        Mockito.when(mockUserRepository.findByUsername("Username")).thenReturn(userTest);
```

Comment:
Should be **marti** the argument
```suggestion
        Mockito.when(mockUserRepository.findByUsername("marti")).thenReturn(userTest);
```